modules/navier_stokes/src/fvbcs/PINSFVStrongBC.C

```diff
@@ -0,0 +1,90 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PINSFVStrongBC.h"
+#include "NS.h"
+#include "SinglePhaseFluidProperties.h"
+#include "Function.h"
+#include "MfrPostprocessor.h"
+
+registerMooseObject("NavierStokesApp", PINSFVStrongBC);
+
+InputParameters
+PINSFVStrongBC::validParams()
+{
+  InputParameters params = FVFluxBC::validParams();
+  params += INSFVFlowBC::validParams();
+  params.addClassDescription("Computes the residual of advective term using finite volume method.");
+  params.addParam<MaterialPropertyName>(NS::density, NS::density, "The density material property");
+  MooseEnum eqn("mass momentum energy");
+  params.addRequiredParam<MooseEnum>("eqn", eqn, "The equation you're solving.");
+  MooseEnum momentum_component("x=0 y=1 z=2");
+  params.addParam<MooseEnum>("momentum_component",
+                             momentum_component,
+                             "The component of the momentum equation that this kernel applies to.");
+  params.addRequiredCoupledVar(NS::superficial_velocity_x,
+                               "The x-component of the superficial velocity");
+  params.addCoupledVar(NS::superficial_velocity_y, "The y-component of the superficial velocity");
+  params.addCoupledVar(NS::superficial_velocity_z, "The z-component of the superficial velocity");
+  params.addRequiredCoupledVar(NS::pressure, "The pressure");
+  params.addRequiredCoupledVar(NS::porosity, "The porosity");
+  return params;
+}
+
+PINSFVStrongBC::PINSFVStrongBC(const InputParameters & params)
```

Comment:
Strong or implicit?
This doesn't take reference values like PCNSFVStrongBC. Feels more implicit to me

Reply:
This can either be implicit or include dirichlet information depending on whether the user puts dirichlet information in their input. I've updated the documentation to indicate this as well as renamed this to `PINSFVFunctorBC`